### PR TITLE
Fix HACKING.md description of how to run tests

### PR DESCRIPTION
**Pull request by lspitzner**
see #2334

### CODE REVIEW

**Comment by tibbe** [comment]
I used to be able to just run `cabal test` after creating a sandbox and have that work. Do we know what changed?

/cc @ttuegel

**Comment by ttuegel** [comment]
@tibbe I have never specifically tried with a sandbox, but I have not been able to use cabal-install to build Cabal since cabal-install started automatically reconfiguring (the persistent build config generated by Cabal's Setup always has a different version). I am investigating this now, because `cabal build` works for @lspitzner, but not `cabal test`. Both commands reconfigure in the same way, so it seems either both should work, or neither. If `cabal build` works, though, there is hope for `cabal test`!

**Comment by lspitzner** [comment]
so.. how do _you_ run the tests? I certainly am curious if there is a simpler way than the one i described; it is too complex for my liking.

**Comment by ttuegel** [comment]
@lspitzner I run tests with `ghc --make Setup; ./Setup configure --enable-tests; ./Setup build; ./Setup test`.

**Comment by dcoutts** [comment]
@ttuegel this is the same issue I think as in the other ticket we were looking at recently with the binary setup-conf file. The test command always reads the setup-conf, and so in cases where the versions differ then it'd also always fail.

**Comment by dcoutts** [comment]
In theory we ought to be able to fix things so that cabal test just works for Cabal's tests.

**Comment by lspitzner** [comment]
oh, i forgot some -Wall. Ping me for the case that this is actually merged.

File: HACKING.md [hunk 1]
```diff
@@ -31,45 +34,93 @@ Building Cabal from from source requires the following:
...
+    .cabal-sandbox/x86_64-linux-ghc-7.8.4-packages.conf.d
+    ~~~~
 
-5. Build cabal-install and run the tests:
+    We will refer to this as `PACKAGEDB`.
+
+5. Configure and build Cabal, and run all tests:
+
+    ~~~~
+    ./Setup configure --enable-tests --package-db=$PACKAGEDB
```
Comment (hdgarrood) [inline HACKING.md]: To avoid accidental CI failures, it could be good to add `--ghc-option=-Wall --ghc-option=-Werror` here.

**Comment by hdgarrood** [comment]
Even if it's only meant to be temporary, I think it would be good to have this merged. I was recently trying to contribute to `Cabal` and it was very painful - this was the only way I could get the build to work. I also don't see any downside to using sandboxes; certainly it's my default approach for working on Haskell packages.

**Comment by ttuegel** [comment]
Merged. I thought it would be simple to fix `cabal test` to just work, but I can't budget the time for it right now.

**Pull request merged**